h2o-genmodel/src/main/java/hex/genmodel/easy/RowToRawDataConverter.java

```diff
@@ -0,0 +1,175 @@
+import java.net.URL;
+import java.util.HashMap;
+
+/**
+ * This class is intended to transform a RowData instance - for which we want to get prediction to - into a raw array
+ */
+public class RowToRawDataConverter {
+
+  public final GenModel _m;
+  private final HashMap<String, Integer> _modelColumnNameToIndexMap;
+  private final HashMap<Integer, HashMap<String, Integer>> _domainMap;
+  private final EasyPredictModelWrapper.ErrorConsumer _errorConsumer;
+
+  private final boolean _convertUnknownCategoricalLevelsToNa;
+  private final boolean _convertInvalidNumbersToNa;
+  
+  public RowToRawDataConverter(GenModel m,
+                               HashMap<String, Integer> modelColumnNameToIndexMap,
+                               HashMap<Integer, HashMap<String, Integer>> domainMap,
+                               EasyPredictModelWrapper.ErrorConsumer errorConsumer,
+                               boolean convertUnknownCategoricalLevelsToNa,
+                               boolean convertInvalidNumbersToNa) {
+    _m = m;
+    _modelColumnNameToIndexMap = modelColumnNameToIndexMap;
+    _domainMap = domainMap;
+    _errorConsumer = errorConsumer;
+    _convertUnknownCategoricalLevelsToNa = convertUnknownCategoricalLevelsToNa;
+    _convertInvalidNumbersToNa = convertInvalidNumbersToNa;
+  }
+
+  /**
+   * 
+   * @param data instance of RowData we want to get prediction for
+   * @param rawData array that will be filled up from RowData instance and returned
+   * @return `rawData` array with data from RowData. 
+   * @throws PredictException Note: name of the exception feels like out of scope of the class with name `RowToRawDataConverter` 
+   * but this conversion is only needed to make it possible to produce predictions so it makes sense
+   */
+  public double[] convert(RowData data, double[] rawData) throws PredictException {  
+
+    // TODO: refactor
+    boolean isImage = _m instanceof DeepwaterMojoModel && ((DeepwaterMojoModel) _m)._problem_type.equals("image");
+    boolean isText  = _m instanceof DeepwaterMojoModel && ((DeepwaterMojoModel) _m)._problem_type.equals("text");
+
+    for (String dataColumnName : data.keySet()) {
+      Integer index = _modelColumnNameToIndexMap.get(dataColumnName);
+
+      // Skip column names that are not known.
+      // Skip the "response" column which should not be included in `rawData`
+      if (index == null || index >= rawData.length) {
+        continue;
+      }
+
+      BufferedImage img = null;
+      String[] domainValues = _m.getDomainValues(index);
+
+      if (domainValues == null) {
+        // Column is either numeric or a string (for images or text)
+        double value = Double.NaN;
+        Object o = data.get(dataColumnName);
+        if (o instanceof String) {
+          String s = ((String) o).trim();
+          // Url to an image given
+          if (isImage) {
+            boolean isURL = s.matches("^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]");
```

Comment:
looks like a terrible idea (and I'm sure many wrong formatted urls match this regex).
Why not ```try { new URL(s); isUrl = true; } catch (Exception ignored) { isUrl = false; }``` instead?

I understand it's legacy, but my eyes still bleeding.

Reply:
For this PR I think it's appropriate to change nothing from the old code except for moving it around.  But for the future, it would be safe to remove the Deep Water related stuff like isImage.

Reply:
@seb-h2o How would you call converter then? I remember the article that you have sent to me about kingdom of Nouns. But I think that the truth is somewhere in the middle. There is such an abstraction as Conversion so I think in this case it is ok to add corresponding suffix. 
Regarding urls regexp... I was trying not to look inside the method :) Let's create separate class URLMatcher :)

Reply:
This is really dead code at this point.  The best way to refactor it is to remove it (but not in this PR).

Reply:
@tomkraljevic do you want me to create a jira for this?

Reply:
sure.

Reply:
I tend to disagree, since the purpose of this PR is to refactor (move around) this code, I would vote to make it better. But I understand you have different fish to fry ATM